django/db/migrations/serializer.py

```diff
@@ -244,8 +244,9 @@ def _format(self):
 
 class SetSerializer(BaseSequenceSerializer):
     def _format(self):
-        # Don't use the literal "{%s}" as it doesn't support empty set
-        return "set([%s])"
+        # When value is empty, the empty set should be serialized as "set()",
+        # not "{}" because {} is an empty dict.
+        return '{%s}' if self.value else 'set(%s)'
```

Comment:
In general I agree. It was only this specific case I had issues with because it introduces a branch which is noisier than the syntax we're trying to standardize on. That said it's in a codegen path, so the nicer output is likely to outweigh messier input. 

No objections to merge.